#7 screen create , edit account

trandinhlam opened the pull request:
#7

Commit:
#7 change login_id to username & access control master/account (only admin have privilege)

Commit:
Merge branch 'production-management-system' into dev/7-list_account_screen_for_admin

Commit:
#7 updated 2nd

Commit:
#7 updated 3rd

Commit:
#7 updated 4rd

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
 	{
 		parent::__construct();
+		$this->force_admin();
+		$this->check_expired_account();
```
@trandinhlam 
Gọi method check_expired_account() trong MY_Controller để áp dụng cho tất cả Controller

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
 
 class Account extends MY_Controller 
 {	
+	protected $per_page = 2;
```
@trandinhlam 
per_page để mặc định, nếu có yêu cầu đặc biệt thì để ở file config.

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
+		$count_by_keyword = $this->accounts_model->count_by_keyword($keyword);
+		if($keyword == "")
+		{
+				$this->load_pagination('/master/account',$count_by_keyword, $this->per_page);
```
@trandinhlam 
đoạn này có thể được gọi, nhưng không bao giờ được áp dụng.

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
+		$page_links = $this->pagination->create_links();
+		$this->data['pagination'] = $page_links;
+		$start = $this->input->get('start');
+		if( $start > $count_by_keyword)
```
@trandinhlam 
Viết thành method get_last_page_start trong MY_Pagination

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
+		$new_account_id = $this->accounts_model->insert_account($this->data['account']);
+		if($new_account_id)
+		{
+			$this->session->set_flashdata('message', "アカウント情報を更新しました");
```
@trandinhlam 
Câu thông báo thành công và thất bại chưa đúng.
thành công: アカウントを作成しました
thất bại: アカウントを作成することができません

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
+		}
+		if($this->accounts_model->get_account_by_id($account_id))
+		{
+			$this->data['account'] = $this->accounts_model->get_account_by_id($account_id);
```
@trandinhlam 
gọi $this->accounts_model->get_account_by_id($account_id) 2 lần

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,189 @@
+		else
+		{
+			$old_info = $this->accounts_model->get_account_by_id($this->data['account']['account_id']);
+			$this->data['account']['password'] = $old_info['password'];
```
@trandinhlam 
trường hợp không thay đổi password thì không cập nhật field password.

alx-github commented on www/codeigniter/application/core/MY_Controller.php:
```diff
@@ -55,12 +66,22 @@ protected function render_menu($current_tab)
 	}
 	
-	protected function load_pagination($url, $count)
+	protected function load_pagination($url, $count,$per_page)
```
@trandinhlam 
$per_page để mặc định, nếu muốn thay đổi phải có giá trị mặc định

alx-github commented on www/codeigniter/application/core/MY_Controller.php:
```diff
@@ -55,12 +66,22 @@ protected function render_menu($current_tab)
 		]);
 	}
+
+	public function username_check($str)
```
@trandinhlam 
Chỉ sử dụng ở màn hình account, nên chuyển sang controller Account

alx-github commented on www/codeigniter/application/models/Accounts_model.php:
```diff
@@ -67,4 +67,88 @@ public function check_username_exists($username)
 		return (count($query->result_array()) > 0);
 	}
+
+	public function count_all()
```
@trandinhlam 
không cần thiết, sử dụng count_by_keyword với keyword null

alx-github commented on www/codeigniter/application/models/Accounts_model.php:
```diff
@@ -67,4 +67,88 @@ public function check_username_exists($username)
+		return $this->db->count_all_results();
+	}
+
+	public function count_by_keyword($keyword)
```
@trandinhlam 
thiết lập giá trị default cho keyword

alx-github commented on www/codeigniter/application/models/Accounts_model.php:
```diff
@@ -67,4 +67,88 @@ public function check_username_exists($username)
+		$this->db->where('account_id',$account_id);
+		$this->db->where('deleted_at',null);
+		$result = $this->db->get();
+		$this->db->trans_complete();
```
@trandinhlam 
không cần thiết

alx-github commented on www/codeigniter/application/models/Accounts_model.php:
```diff
@@ -67,4 +67,88 @@ public function check_username_exists($username)
+		return $result->result_array()[0];
+	}
+
+	public function insert_account($account)
```
@trandinhlam 
insert, update, delete sử dụng Base_model

alx-github commented on www/codeigniter/application/views/master/account/form.php:
```diff
@@ -6,63 +6,69 @@
-							<option>縫製ユーザー</option>
-							<option>出荷ユーザー</option>
+						<select class="form-control" name="auth">
+						<?php if (($account['auth']) === null): ?>
```
@trandinhlam 
không cần if-else
tạo account_auth trong config
$config['account_auth'] = [$key => $value]

alx-github commented on www/codeigniter/application/views/master/account/list.php:
```diff
@@ -14,31 +34,15 @@
-					<span aria-hidden="true">&raquo;</span>
-					</a></li>
-			</ul>
+			<?php echo $pagination ?>
```
@trandinhlam 
sử dụng method của framework

alx-github commented on www/codeigniter/application/views/master/account/list.php:
```diff
@@ -48,57 +52,46 @@
-					</tr>
-						<td>user1</td>
-						<td>一般ユーザー</td>
+						<td><?=$account['username']?></td>
```
@trandinhlam 
sử dụng $config['account_auth'] trong config

alx-github commented on www/codeigniter/application/views/master/account/list.php:
```diff
@@ -48,57 +52,46 @@
-					<span aria-hidden="true">&raquo;</span>
-					</a></li>
-			</ul>
+			<?php echo $pagination ?>
```
@trandinhlam 
sử dụng method của framework

Commit:
#7  updated 5th:
*check_expired_account move to MY_controller
*pagination: split into "get_last_page_start" in MY_pagination
*pagination: set $per_page to default = 10
* update message for creating account
*set "keyword" default null
*only update new password if need
*use Base_model for insert,update,delete
*use $config['account_auth'] for select form

Commit:
#7  updated 6th:
* set seleted in edit screen
*pagination: return value in "get_last_page_start" function

Commit:
#7  updated 7th:
*Base_model: update function get_by_id, insert, update, delete
*add $primary_key_name
*remove functions (insert,update,delete) in Account_model

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,173 @@
 	public function index()
 	{
+		$keyword = null;
+		if($this->input->get('keyword'))
```
@trandinhlam 
không cân if vì không tìm thấy thì input->get sẽ trả ra NULL.

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,173 @@
+		}
+		$count_by_keyword = $this->accounts_model->count_by_keyword($keyword);
+		$this->load_pagination('/master/account?keyword='.$keyword,$count_by_keyword);
+		$start = $this->pagination->get_last_page_start($this->input->get('start'),$count_by_keyword);
```
@trandinhlam 
lấy $start từ $this->input
trường hợp $start > $this->pagination->get_last_page_start() thì thay đổi $start

alx-github commented on www/codeigniter/application/controllers/master/Account.php:
```diff
@@ -3,48 +3,173 @@
+        return true;
+	}
+
+	public function username_check($str)
```
@trandinhlam 
chuyền thành private

alx-github commented on www/codeigniter/application/libraries/My_pagination.php:
```diff
@@ -267,4 +267,25 @@ class My_Pagination extends CI_Pagination {
 	 */
 	protected $data_page_attr = 'data-ci-pagination-page';
+
+	public function get_last_page_start($input, $count_by_keyword)
```
@trandinhlam 
không cần tham số.
sau khi load_pagination(), last_page_start sẽ không thay đổi.
sử dụng $this->total_rows và $this->per_page để tính toán

alx-github commented:
@trandinhlam 
thay đổi toàn bộ 
null -> NULL
if () : endif;  -> if () {}
for () : endfor; -> for () {}
style : end...; -> () {}

Commit:
#7  updated 8th:
*change style : end... => () {}
*remove parameters in "get_last_start_page" function

alx-github commented on www/codeigniter/application/libraries/My_pagination.php:
```diff
@@ -49,7 +49,7 @@ class My_Pagination extends CI_Pagination {
 	 * @var	int
 	 */
-	public $per_page = 10;
+	public $per_page = 2;
```
@trandinhlam 
Mặc định là 10.

Commit:
#7  updated 8th:
*change per_page  = 10
